src/hotspot/share/nmt/mallocTracker.cpp

```diff
@@ -218,96 +216,3 @@ void MallocTracker::deaccount(MallocHeader::FreeInfo free_info) {
     MallocSiteTable::deallocation_at(free_info.size, free_info.mst_marker);
   }
 }
-
-// Given a pointer, look for the containing malloc block.
-// Print the block. Note that since there is very low risk of memory looking
-// accidentally like a valid malloc block header (canaries and all) so this is not
-// totally failproof and may give a wrong answer. It is safe in that it will never
-// crash, even when encountering unmapped memory.
-bool MallocTracker::print_pointer_information(const void* p, outputStream* st) {
```

Comment:
It looks like the only thing that had to go was `candidate->looks_valid()`

Isn't it useful to keep the rest of this code though?

Reply:
We've also removed the `is_live()` call. So now we really can't make a reasonable guess whether a potential pointer really points to a `MallocHeader`. Since, in my opinion, we can't find a reasonable candidate (not sufficient clues), I decided to remove the code.